### ISSUE TITLE
[build] release build with cuda/gdrcopy support fails

### ISSUE THREAD

**Issue by amaslenn** [openucx/ucx#2166]
Checked on 0aa5ed9.

Reproduction:
```sh
$ module load dev/cuda
$ module load dev/gdrcopy
$ ./contrib/configure-release --with-cuda --with-gdrcopy
$ make -j
```

Produces following output:
```sh
CC       sm/knem/libuct_la-knem_md.lo
cuda/gdr_copy/gdr_copy_md.c: In function ‘uct_gdr_copy_md_open’:
cuda/gdr_copy/gdr_copy_md.c:421:36: error: passing argument 3 of ‘ucs_rcache_create’ from incompatible pointer type [-Werror]
                                    UCS_STATS_ARG(NULL), &md->rcache);
                                    ^
In file included from cuda/gdr_copy/gdr_copy_md.h:10:0,
                 from cuda/gdr_copy/gdr_copy_md.c:6:
/auto/mtrswgwork/andreyma/ucx/src/ucs/sys/rcache.h:146:14: note: expected ‘struct ucs_stats_node_t *’ but argument is of type ‘struct ucs_rcache_t **’
 ucs_status_t ucs_rcache_create(const ucs_rcache_params_t *params, const char *name,
              ^
cuda/gdr_copy/gdr_copy_md.c:421:36: error: too few arguments to function ‘ucs_rcache_create’
                                    UCS_STATS_ARG(NULL), &md->rcache);
                                    ^
In file included from cuda/gdr_copy/gdr_copy_md.h:10:0,
                 from cuda/gdr_copy/gdr_copy_md.c:6:
/auto/mtrswgwork/andreyma/ucx/src/ucs/sys/rcache.h:146:14: note: declared here
 ucs_status_t ucs_rcache_create(const ucs_rcache_params_t *params, const char *name,
              ^
cc1: all warnings being treated as errors
make[2]: *** [cuda/gdr_copy/libuct_la-gdr_copy_md.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory `/auto/mtrswgwork/andreyma/ucx/src/uct'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/auto/mtrswgwork/andreyma/ucx'
make: *** [all] Error 2
```

**Comment by yosefe** [comment]
@bureddy pls add jenkins test which builds in release mode with cuda

**Comment by yosefe** [comment]
fixed by #2169
